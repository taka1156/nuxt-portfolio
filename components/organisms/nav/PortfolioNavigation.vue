<template>
  <div>
    <nav>
      <nav-bar :logo-text="logoText" :is-open="isOpen" @change-state="changeState" />
      <nav-list :is-open="isOpen" :routes="routes" @change-state="changeState" />
    </nav>
  </div>
</template>

<script>
import NavBar from '../../molecules/nav/NavBar';
import NavList from '../../molecules/nav/NavList';

export default {
  name: 'BaseNav',
  components: {
    'nav-bar': NavBar,
    'nav-list': NavList,
  },
  props: {
    logoText: {
      type: String,
      default: 'Please Setting Title',
      required: true,
    },
    routes: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  data() {
    return {
      isOpen: false,
    };
  },
  methods: {
    changeState() {
      // ナビゲーションバーの開閉管理
      this.isOpen = !this.isOpen;
    },
  },
};
</script>
