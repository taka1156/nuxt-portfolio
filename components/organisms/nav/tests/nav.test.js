import { mount, RouterLinkStub } from '@vue/test-utils';
import PortfolioNavigation from '../PortfolioNavigation.vue';

// PortfolioNavigation
describe('PortfolioNavigation', () => {
  const dummyLogo = 'ダミー';
  const dummyRoutes = [
    { name: 'Top', to: '/home', img: 'http://placehold.jp/150x150.png' },
  ];

  const baseNav = mount(PortfolioNavigation, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: {
      logoText: dummyLogo,
      routes: dummyRoutes,
    },
  });

  it('NavBar初期値: logoText, routes', () => {
    // logoText
    expect(baseNav.vm.$options.props.logoText.required).toBe(true);
    expect(baseNav.vm.logoText).toBe(dummyLogo);
    // routes
    expect(baseNav.vm.$options.props.routes.required).toBe(true);
    console.log(baseNav.vm.$options.props.routes.default());
    expect(baseNav.vm.routes).toBe(dummyRoutes);
  });

  it('閉じた時のスナップショット', () => {
    expect(baseNav.html()).toMatchSnapshot();
  });

  it('changeStateのテスト', () => {
    expect(baseNav.vm.$data.isOpen).toBe(false);
    baseNav.vm.changeState();
    expect(baseNav.vm.$data.isOpen).toBe(true);
  });

  it('開いた時のスナップショット', () => {
    expect(baseNav.html()).toMatchSnapshot();
  });
});
