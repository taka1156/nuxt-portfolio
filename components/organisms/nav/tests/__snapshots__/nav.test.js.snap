// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioNavigation 閉じた時のスナップショット 1`] = `
"<div>
  <nav>
    <div>
      <div class=\\"nav-bar\\">
        <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav-logo\\">ダミー</a>
          <div>
            <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--close\\"></span> <span class=\\"nav-icon__middle\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--close\\"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <transition-stub name=\\"nav-fade\\">
        <div class=\\"nav-list\\" style=\\"display: none;\\">
          <ul>
            <div>
              <li class=\\"nav-list-item\\"><a class=\\"nav-list-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav-img\\">
                  <p class=\\"base-text base-text--nav\\">Top</p>
                </a></li>
            </div>
          </ul>
        </div>
      </transition-stub>
    </div>
  </nav>
</div>"
`;

exports[`PortfolioNavigation 開いた時のスナップショット 1`] = `
"<div>
  <nav>
    <div>
      <div class=\\"nav-bar\\">
        <div class=\\"nav-bar__box\\"><a href=\\"/\\" class=\\"nav-logo\\">ダミー</a>
          <div>
            <div class=\\"nav-icon\\"><span class=\\"nav-icon__top nav-icon__top--open\\"></span> <span class=\\"nav-icon__middle nav-icon__middle--fade\\"></span> <span class=\\"nav-icon__bottom nav-icon__bottom--open\\"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <transition-stub name=\\"nav-fade\\">
        <div class=\\"nav-list\\" style=\\"\\">
          <ul>
            <div>
              <li class=\\"nav-list-item\\"><a class=\\"nav-list-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"Topの画像\\" class=\\"nav-img\\">
                  <p class=\\"base-text base-text--nav\\">Top</p>
                </a></li>
            </div>
          </ul>
        </div>
      </transition-stub>
    </div>
  </nav>
</div>"
`;
